.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,36 @@
+name: Validate Pull Request
+
+on: [push, pull_request]
```

Comment:
```suggestion
on: [pull_request]
```

Reply:
IINM, "push" is there to help developers working in forks to be able to run these in GitHub actions (This is done in some of the other standard library modules as well I believe).

We can add a filter to only run `build-main` workflow in ballerina-lang main branch and this on all the other branches.

Reply:
IMHO, that should be another workflow. Because, we already have a workflow to run `on push` in `main` branch. Then usually internal developers tends to use the original repo (not the forks) to develop. In that case, each PR will trigger two workflows, one `on push` and one `on pull_request`.

So, if we need a workflow to run on push, we should have a separate workflow with `on push`, ignoring the `main`. 

But anyways, this isn't a major issue. Maybe a subjective one. We can go ahead with this.

Reply:
Yeah that will be better. Shall I create another "build-dev-branches" workflow ?

Reply:
Added new workflow in bc2ed03e3caf8bf290cd97bbeaa84c736d0c08a4